Tests/SwiftPhoenixClientTests/Serializer/SerializerTest.swift

```diff
@@ -91,37 +91,45 @@ final class SerializerTest {
         #expect(message.payloadString == "{\"foo\":\"bar\"}")
     }
     
+    private struct NumberMessage: Codable {
+        let int: Int
+        let float: Float
+    }
+    
     @Test func test_decodeMessageWithNumbers() throws {
         let text = """
-        ["1","2", "topic","event",{"int":1,"float":1.1}]
+        ["1","2", "topic","event",{"int":1234,"float":1.1234}]
         """
         
         let message = try serializer.decode(text: text)
+        let result = try JSONDecoder().decode(NumberMessage.self, from: message.payload)
         
-        let jsonResult = message.payloadString
-        let excpetedResultsContainJsonResult = [
-            "{\"int\":1,\"float\":1.1}",
-            "{\"float\":1.1,\"int\":1}"
-        ].contains(jsonResult)
-        
-        #expect(excpetedResultsContainJsonResult)
+        #expect(result.int == 1234)
+        #expect(result.float == 1.1234)
+    }
+    
+    private struct ValuePayload: Codable {
+        let value: String
     }
     
     @Test func test_decodeMessageWithoutJsonPayload() throws {
         // NOTE: Since NSNumber can be a float or an int, `1.0` will convert to `1`
         // but should be able to map `1` back into a float property.
         let text = """
-        ["1","2", "topic","event","payload"]
+        ["1","2", "topic","event",{"value": "payload"}]
```

Comment:
Just because the phoenix docs state that payload _should_ be a map, there is no requirement that it _has to be_ a map. We'll want to use `fragmentsAllowed` in our `options`.

The test values in the class were specifically chosen to handle edge cases and shouldn't be changed, otherwise we aren't guaranteeing that we are handling these edge cases.

Reply:
Got it -- I added a couple commits that turn on `.fragmentsAllowed` and put the test values back as original.